core/src/main/java/com/linecorp/armeria/server/http/auth/OAuth1aToken.java

```diff
@@ -0,0 +1,212 @@
+package com.linecorp.armeria.server.http.auth;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+/**
+ * The bearer token of <a href="https://oauth.net/core/1.0a/#anchor12">OAuth 1.0a authentication</a>.
+ * */
+public final class OAuth1aToken {
+
+    private static final Joiner.MapJoiner mapJoiner = Joiner.on(", ").withKeyValueSeparator("=");
+
+    /**
+     * Creates a new {@link OAuth1aToken} from given arguments.
+     * */
+    public static OAuth1aToken of(Map<String, String> params) {
+        return new OAuth1aToken(params);
+    }
+
+    /**
+     * realm parameter. (optional)
+     * */
+    private final String realm;
+
+    /**
+     * oauth_consumer_key parameter.
+     * */
+    private final String consumerKey;
+
+    /**
+     * oauth_token parameter.
+     * */
+    private final String token;
+
+    /**
+     * oauth_signature_method parameter.
+     * */
+    private final String signatureMethod;
+
+    /**
+     * oauth_signature parameter.
+     * */
+    private final String signature;
+
+    /**
+     * oauth_timestamp parameter.
+     * */
+    private final Long timestamp;
+
+    /**
+     * oauth_nonce parameter.
+     * */
+    private final String nonce;
+
+    /**
+     * version parameter. (optional)
+     * If not set, the default value is 1.0.
+     * */
+    private final String version;
+
+    /**
+     * Additional Parameters.
+     * */
+    private final Map<String, String> additional;
+
+    private OAuth1aToken(Map<String, String> params) {
+        String realm = params.remove("realm");
```

Comment:
How about create `private final Map<String, String> params` field and set `params` argument to params field instead of modifying `params` argument?
If a user call `of(Collections.unmodifiableMap())`, it causes UnsupportedOperationException.